ext/opcache/jit/zend_jit.c

```diff
@@ -2818,6 +2820,18 @@ static int zend_real_jit_func(zend_op_array *op_array, zend_script *script, cons
 	/* Build SSA */
 	memset(&ssa, 0, sizeof(zend_ssa));
 
+	if (op_array->fn_flags & ZEND_ACC_CLOSURE) {
+		if (JIT_G(trigger) == ZEND_JIT_ON_FIRST_EXEC) {
+			zend_jit_op_array_extension *jit_extension = (zend_jit_op_array_extension*)ZEND_FUNC_INFO(op_array);
+			op_array = (zend_op_array*) jit_extension->op_array;
+		} else if (JIT_G(trigger) == ZEND_JIT_ON_HOT_COUNTERS) {
+			zend_jit_op_array_hot_extension *jit_extension = (zend_jit_op_array_hot_extension*)ZEND_FUNC_INFO(op_array);
+			op_array = (zend_op_array*) jit_extension->op_array;
+		} else {
+			ZEND_ASSERT(!op_array->scope);
```

Comment:
What about ``ZEND_FUNC_JIT_ON_PROF_REQUEST``?

Reply:
I added it here initially, but I then I figured that it was not necessary because we never compile closures in `ZEND_FUNC_JIT_ON_PROF_REQUEST` mode (according to my understanding of `zend_jit_check_funcs()`). In theory we could compile closures in `zend_jit_check_funcs()` by also iterating over `op_array->dynamic_func_defs`, but in this case we would get the right `op_array`.

Reply:
OK.